lib/improver/tests/utilities/test_pad_spatial.py

```diff
@@ -173,6 +174,45 @@ def test_values(self):
                                     self.cube.coord(axis='y').points)
 
 
+class Test_remove_cube_halo(IrisTest):
+    """Tests for the create_cube_with_halo function"""
+
+    def setUp(self):
+        """Set up a realistic input cube with lots of metadata.  Input cube
+        grid is 1000x1000 km with points spaced 100 km apart."""
+        attrs = {'history': '2018-12-10Z: StaGE Decoupler',
+                 'title': 'Temperature on UK 2 km Standard Grid',
+                 'source': 'Met Office Unified Model'}
+
+        self.cube = set_up_variable_cube(
+            np.ones((3, 11, 11), dtype=np.float32), spatial_grid='equalarea',
+            standard_grid_metadata='uk_det', attributes=attrs)
+
+        self.cube_1d = set_up_variable_cube(
+            np.ones((1, 11, 11), dtype=np.float32), spatial_grid='equalarea',
+            standard_grid_metadata='uk_det', attributes=attrs)
+        self.grid_spacing = 100000
+
+    def test_basic(self):
+        """Test function returns a cube with expected metadata and shape"""
+        halo_size_km = 162.
+        result = remove_cube_halo(self.cube, 1000.*halo_size_km)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertTrue(result.attributes)
+        self.assertSequenceEqual(result.data.shape, (3, 9, 9))
```

Comment:
Probably need to show cell methods are preserved too.  I don't think `set_up_variable_cubes` does this, so feel free to add functionality if you want, or just add cell methods to the cubes in your local `setUp` function.

Reply:
Since the cell methods and attributes are all inherited in _create_cube_with_padded_data I shall remove this test.

Reply:
Actually I'd recommend we keep this test.  If changes are ever made to `_create_cube_with_padded_data` such that attributes / cell-methods are not preserved, we would need this test to fail so that we know the impact of those changes.

Reply:
Surely there are unit tests in _create_cube_with_padded_data which check these are preserved?